src/common/offline_transformations/include/mask_attribute.hpp

```diff
@@ -103,7 +103,20 @@ class Mask : public std::vector<std::set<uint64_t>>,
         }
     }
 
-    Mask::Ptr intersect_masks_reversed(Mask *const mask) {
+    void copy_value_from_mask_reversed_masked(Mask *const mask, std::set<int64_t> idx_mask, const bool invert_mask = false) {
+        auto cur_mask_iter = rbegin();
+        auto mask_iter = mask->rbegin();
+        while (cur_mask_iter != rend() && mask_iter != mask->rend()) {
```

Comment:
what if the mask lengths are not equal? is it mistake? should we throw exception or assert in this case?

and, as I can see, Mask class is inherited from std::vector, so maybe it's better to use indexes [idx], not iterators rbegin, rend)

Reply:
1) This approach is intended to work with different masks lengths in case like bias add:
     (N, C, H, W)        (C, H, W)
     |                                |
      \                              /
                 (add)
                 
  So, we match masks dimensions starting from the end
  
 2) And with such approach use of backward iterators looks more clear from my perspective (instead of index calculation for each operand)

Reply:
discussed offline, it looks correct

Reply:
Comments was added